RadioIDs.py

```diff
@@ -27,14 +32,21 @@ def _scrape(sources: List[str]) -> List[str]:
         List[str]: [description]
     """
     names: List[str] = []
+    toLookup: List[str] = []
     for source in sources:
         if source in namecache.keys():
             names.append(namecache[source])
         else:
-            # API stuff here, in progress. Even cache none so the API isn't battered
-            name = None
-            namecache[source] = name
-            names.append(name)
+            toLookup.append(source)
+
+    if toLookup:
+        response = requests.get(LOOKUP_URL, params={"radio": toLookup})
+        data = response.json()
+        log.debug(f"Data back from API: {data}")
+        for source, info in data.items():
+            namecache[str(source)] = info
+            names.append(info)
+
     return [name for name in names if name is not None]
```

Comment:
I think this can just be:
```suggestion
    return names
```

Reply:
Incorporated, but had to make sure to filter in bot ;)